### PR TITLE
Get preferred UI languages from the LANGUAGE variable on *nix.

### PR DESCRIPTION
**Pull request by lanurmi**
Obey the LANGUAGE environment variable (if set), which is GNU
gettext's primary way of determining language preference. Apparently
ignored by wx until now, even though wx attempts to be a
reimplementation of GNU gettext.

The LANGUAGE variable may contain a list of preferred languages,
so use that list to find the best translation.

GNU gettext has supported multiple preferred languages since forever
(at least 15 years), and therefore it is odd that this wasn't already
implemented in 01f953efb211daf5275cbb10aa3dd000eeed6d48. Instead,
it was implied that Unix does not support such. (Even if GNU is not
Unix, nothing stops a wxWidgets app supporting such nevertheless.)

### CODE REVIEW

File: src/common/translation.cpp [hunk 1]
```diff
@@ -125,6 +125,54 @@ wxString GetPreferredUILanguageFallback(const wxArrayString& WXUNUSED(available)
...
+        {
+            const wxString tok = tknzr.GetNextToken();
+            if ( const wxLanguageInfo *li = wxLocale::FindLanguageInfo(tok) )
+            {
+                preferred.push_back(li->CanonicalName);
+            }
+        }
+        if ( preferred.empty() )
+        {
+            wxLogTrace(TRACE_I18N, " - LANGUAGE was set, but it didn't contain existing languages");
```
Comment (vadz) [inline src/common/translation.cpp]: The wording here is a bit weird, I think "recognized" would be more clean than "existing" (if I understand correctly what is this trying to say).

**Review by vadz — APPROVED** [review]
Looks good to me, thanks!

@vslavik Would you have any thoughts about this by chance? If not, I'll merge this soon.

File: src/common/translation.cpp [hunk 2]
```diff
@@ -248,9 +296,9 @@ wxString GetPreferredUILanguage(const wxArrayString& available)
 
 #else
 
-// On Unix, there's just one language=locale setting, so we should always
-// use that.
-#define GetPreferredUILanguage GetPreferredUILanguageFallback
+// On *nix, a list of preferred languages may be specified in the LANGUAGE
+// environment variable.
+#define GetPreferredUILanguage GetPreferredUILanguageFromEnvVariable
```
Comment (vslavik) [inline src/common/translation.cpp]: The whole `GetPreferredUILanguageFromEnvVariable()` function should be just called `GetPreferredUILanguage()` and its body moved to here, getting rid of the `#define` altogether. Its name is misleading anyway (the *Fallback one uses environment variable too) and it is only used on UNIX, so it doesn't make sense to compile it on other platforms and to obfuscate the code with such renamings.

**Comment by vslavik** [comment]
No objections from me (aside from the above comment about code organization). Apparently, `LANGUAGE` actually is used by e.g. modern versions of GNOME, although it took some non-trivial fiddling with language settings for it to set the variable (it didn't by default) — so this definitely is a welcome improvement.

File: src/common/translation.cpp (hunk 1 above)
Comment (lanurmi) [reply src/common/translation.cpp]: I'll rephrase it as "... didn't contain any languages recognized by the system" or something like that.

**Comment by vadz** [comment]
Merged, thanks again!

**Pull request closed without merging**